x/nexus/keeper/general_message.go

```diff
@@ -71,3 +97,18 @@ func (k Keeper) getMessages(ctx sdk.Context) (generalMessages []exported.General
 
 	return generalMessages
 }
+
+// ConsumeApprovedMessages returns up to limit messages where chain is the destination chain, and deletes the messages from the store
+func (k Keeper) ConsumeApprovedMessages(ctx sdk.Context, chain exported.ChainName, limit int64) []exported.GeneralMessage {
```

Comment:
What duplication? And why is it EVM specific?

Reply:
Have a separate table now that stores approved message IDs. EVM specific because only evm chains will consume general messages in the end blocker. For cosmos chains, IBC transfer is triggered by an sdk message.